### PR TITLE
binary serialization for collection summary

### PR DESCRIPTION
**Pull request by eishay**
@ymatsuda @raymondng @andrewconner @stkem someone please review

### CODE REVIEW

**Comment by stkem** [comment]
lgtm

**Comment by raymondng** [comment]
Thanks. This should help a lot!

File: kifi-backend/modules/common/app/com/keepit/model/Collection.scala [hunk 1]
```diff
@@ -45,6 +50,57 @@ object Collection {
...
+    // Not sure it would worth it to do an accurate calculation, to be revised
+    val byteStream = new ByteArrayOutputStream(60 * collections.size)
+
+    byteStream.write(prefix)
+    val outStream = new DataOutputStream(byteStream)
+
+    collections foreach { collection =>
+      outStream.writeLong(collection.id.id) //2 bytes
+      outStream.write(collection.externalId.id.getBytes(UTF8)) //ExternalIdByteSize bytes
+      val name = collection.name.getBytes(UTF8)
```
Comment (ymatsuda) [inline kifi-backend/modules/common/app/com/keepit/model/Collection.scala:69]: Is there any reason not to use writeUTF/readUTF?

Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/model/Collection.scala:69]: I didn't know read/write UTF reads/writes the string length!
Good idea @ymatsuda updating the code

**Pull request merged**